sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs

```diff
@@ -28,9 +32,13 @@ public JsonObjectSerializer() : this(new JsonSerializerOptions())
         /// Initializes new instance of <see cref="JsonObjectSerializer"/>.
         /// </summary>
         /// <param name="options">The <see cref="JsonSerializerOptions"/> instance to use when serializing/deserializing.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="options"/> is null.</exception>
         public JsonObjectSerializer(JsonSerializerOptions options)
         {
-            _options = options;
+            _options = options ?? throw new ArgumentNullException(nameof(options));
```

Comment:
Why throw?  Could we default or leave null?

Reply:
It's a developer error, not a runtime error. If they want the default, they should either use the default constructor above or pass it in here. Also discussed this with @pakrym. We need a default to get the right default naming policy.